### PR TITLE
release: add check for gh auth status

### PR DESCRIPTION
**Pull request by BolajiOlajide**
### Checklist

- [ ] Follow the [manual testing process](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/TEST.md)
- [ ] Update [changelog](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/charts/sourcegraph/CHANGELOG.md)
- [ ] Update [Kubernetes update doc](https://docs.sourcegraph.com/admin/updates/kubernetes)

### Test plan

Add step to check that release captain is authenticated via `gh` during the release process.

### CODE REVIEW

**Review by jdpleiness — APPROVED** [review]
Nice!

**Review by DaedalusG — APPROVED** [review]
Nice I believe @Chickensoupwithrice ran into this

File: release.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ internal:
   create:
     steps:
       patch:
+          - name: "gh:auth"
+            cmd: |
+              gh auth status
```
Comment (jhchabran) [inline release.yaml:26]: I think we should instead move this to the `requirements` and instead call `checkDeps` in `(*releaseRunner) Test(context.Context) error`. The repetition here is very error prone.

**Review by jhchabran — CHANGES_REQUESTED** [review]

File: release.yaml (hunk 1 above)
Comment (jhchabran) [reply release.yaml:26]: https://github.com/sourcegraph/sourcegraph/pull/62143

**Review by jhchabran — APPROVED** [review]
LGTM, I'll fix the part where we need to run this in the tests, over https://github.com/sourcegraph/sourcegraph/pull/62143

**Pull request merged**